src/anyio/_core/_subprocesses.py

```diff
@@ -75,3 +87,103 @@ async def open_process(command: Union[str, Sequence[str]], *, stdin: int = PIPE,
+    try:
+        workers = _process_pool_workers.get()
+        idle_workers = _process_pool_idle_workers.get()
+    except LookupError:
+        workers = set()
+        idle_workers = set()
+        _process_pool_workers.set(workers)
+        _process_pool_idle_workers.set(idle_workers)
+        get_asynclib().setup_process_pool_exit_at_shutdown(workers)
+
+    async with (limiter or current_default_worker_process_limiter()):
+        with open_cancel_scope(shield=not cancellable):
+            if idle_workers:
+                process = idle_workers.pop()
+                stdout = cast(ByteReceiveStream, process.stdout)
+            else:
+                command = [sys.executable, '-u', '-m', 'anyio._core._subprocess_worker']
+                process = await open_process(command, stdin=subprocess.PIPE,
+                                             stdout=subprocess.PIPE)
+                stdout = cast(ByteReceiveStream, process.stdout)
+                with fail_after(20):
+                    message = await stdout.receive(6)
+
+                if message != b'READY\n':
+                    process.kill()
+                    raise BrokenWorkerProcess(
+                        f'Worker process returned unexpected response: {message!r}')
+                else:
+                    workers.add(process)
+
+            stdin = cast(ByteSendStream, process.stdin)
+            try:
+                # Send the run command
+                await stdin.send(request)
+
+                # Receive the response
+                buffered = BufferedByteReceiveStream(stdout)
+                response = await buffered.receive_until(b'\n', 50)
+                status, length = response.split(b' ')
+                if status not in (b'RETURN', b'EXCEPTION'):
+                    raise RuntimeError('Worker process returned unexpected response:',
+                                       response)
+
+                pickled = await buffered.receive_exactly(int(length))
+            except BaseException as exc:
+                workers.discard(process)
+                try:
+                    process.kill()
+                    await process.wait()
+                except ProcessLookupError:
+                    pass
+
+                raise BrokenWorkerProcess from exc
+
+            idle_workers.add(process)
+            retval = pickle.loads(pickled)
+            if status == b'EXCEPTION':
+                assert isinstance(retval, BaseException)
+                raise retval
+            else:
+                return retval
+
+
+def current_default_worker_process_limiter() -> CapacityLimiter:
+    """
+    Return the capacity limiter that is used by default to limit the number of worker processes.
+
+    :return: a capacity limiter object
+
+    """
+    try:
+        return _default_process_limiter.get()
+    except LookupError:
+        limiter = create_capacity_limiter(5)
```

Comment:
why don't use `os.cpu_count()` here? why 5?

Reply:
Right, I intended to change this but got caught up in more pressing matters.